src/hbbft.erl

```diff
@@ -226,7 +232,7 @@ handle_msg(Data = #hbbft_data{round=R}, J, {{acs, R}, ACSMsg}) ->
                                         {multicast, {dec, Data#hbbft_data.round, I, SerializedShare}}
                                 end, Results),
             %% verify any shares we received before we got the ACS result
-            VerifiedShares = maps:map(fun({I, _}, {undefined, Share}) ->
+            VerifiedShares = maps:map(fun({I, _}, {undefined, Share}) when I /= J ->
```

Comment:
We actually need to check if we have a copy of our own proposal here in case of an upgrade